modules/tracker/controllers/components/ApiComponent.php

```diff
@@ -370,18 +368,137 @@ public function submissionValidate($args)
+                    $producerDao,
+                    $user,
+                    MIDAS_POLICY_WRITE
+                )) {
+                    throw new Exception('Write permission on the producer required', 403);
+                }
+                /** @var stdClass $producerDefinition */
+                $producerDefinition = json_decode($producerConfig);
+                // Ensure that Producer and Community match.
+                if ($producerDefinition->producer !== $producerDao->getDisplayName()) {
+                    throw new Exception('Producer schema name must match existing Producer display name', 404);
+                }
+                if ($producerDefinition->community !== $producerDao->getCommunity()->getName()) {
+                    throw new Exception('Producer schema community name must match existing Producer Community name', 404);
+                }
+
+                // Save the producer definition to the producer.
+                $producerDao->setProducerDefinition($producerConfig);
+                // Update top level fields on the producer based on the definition.
+                if (isset($producerDefinition->histogram_max_x)) {
+                    $producerDao->setHistogramMaxX($producerDefinition->histogram_max_x);
+                }
+                if (isset($producerDefinition->grid_across_metric_groups)) {
+                    $producerDao->setGridAcrossMetricGroups($producerDefinition->grid_across_metric_groups);
+                }
+                if (isset($producerDefinition->histogram_number_of_bins)) {
+                    $producerDao->setHistogramNumberOfBins($producerDefinition->histogram_number_of_bins);
+                }
+                $producerModel->save($producerDao);
+
+                $defaults = $producerDefinition->defaults;
+                if (!isset($defaults)) {
+                    // Provide a default for $defaults so the below ?: logic works.
+                    $defaults = new stdClass();
+                }
+                // Add or update any key metrics and thresholds.
+                /** @var Tracker_TrendModel $trendModel */
+                $trendModel = MidasLoader::loadModel('Trend', 'tracker');
+                /** @var Tracker_TrendThresholdModel $trendThresholdModel */
+                $trendThresholdModel = MidasLoader::loadModel('TrendThreshold', 'tracker');
+                $keyMetrics = $producerDefinition->key_metrics;
+                /** @var stdClass $keyMetric */
+                foreach ($keyMetrics as $keyMetric) {
+                    // Set any needed trends to be key_metrics.
+                    $trendModel->setAggregatableTrendAsKeyMetrics($producerDao, $keyMetric->name);
+                    $trendThresholdModel->upsert($producerDao, $keyMetric->name,
+                        isset($keyMetric->abbreviation) ? $keyMetric->abbreviation : false,
+                        isset($keyMetric->warning) ? $keyMetric->warning :
```

Comment:
I have a fear of nested ternaries (I was bitten as a child). Could you create a more clear helper function and use that here rather than the nested ternaries?